integration-tests/maven/src/test/java/io/quarkus/maven/it/DevMojoIT.java

```diff
@@ -484,6 +484,25 @@ public void testThatApplicationRecoversStartupIssue() throws MavenInvocationExce
         assertThat(last.get()).isEqualTo("hello");
     }
 
+    @Test
+    public void testThatResourcesDeletionArePickedUpBeforeFirstRequest() throws MavenInvocationException, IOException {
+        testDir = initProject("projects/classic", "projects/project-classic-run-deletion-of-resources-before-first-request");
+
+        // Create the nresource
+        File source = new File(testDir, "src/main/resources/META-INF/resources/lorem.txt");
+        FileUtils.write(source,
+                "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.",
+                "UTF-8");
+
+        // run the application
+        run();
+
+        // delete the resource before first request
+        source.delete();
+
+        assertThat(getHttpResponse("/lorem.txt", 404)).isEqualTo(true); // verify that index file has been deleted
+    }
```

Comment:
Could we avoid adding a test here and use the new dev mode test infrastructure? The idea is to get rid of most of this file at some point.

Reply:
Thanks for pointing this out. I am currently checking it out.

Reply:
Done